packages/core/util/offscreenCanvas/Canvas2DContextShim/context.ts

```diff
@@ -0,0 +1,211 @@
+  ...arg0: any[]
+) => any
+  ? Parameters<OffscreenCanvasRenderingContext2DShim[METHODNAME]>
+  : never
+
+/** decode all the commands in the given buffer and replay them onto the given context */
+export function replayCommandsOntoContext(
+  targetContext: CanvasRenderingContext2D,
+  encodedCommands: Iterable<Command>,
+) {
+  for (const command of encodedCommands) {
+    if (isMethodCall(command)) {
+      // @ts-ignore
+      // eslint-disable-next-line prefer-spread
+      targetContext[command.name].apply(targetContext, command.args)
+    } else {
+      // @ts-ignore
+      targetContext[command.name] = command.args[0]
+    }
+  }
+}
+
+export default class OffscreenCanvasRenderingContext2DShim {
+  width: number
+  height: number
+
+  currentFont = '12px Courier New, monospace'
+  currentStrokeStyle = ''
+  currentFillStyle = ''
+
+  recordedCommands: Command[] = []
+
+  constructor(width: number, height: number) {
+    this.width = width
+    this.height = height
+  }
+
+  private pushMethodCall(name: MethodName, args: unknown[]) {
+    this.recordedCommands.push({ name, args })
+  }
+
+  private pushSetterCall(name: SetterName, arg: unknown) {
+    this.recordedCommands.push({ name, args: [arg] })
+  }
+
+  getSerializedCommands() {
+    return this.recordedCommands
+  }
+
+  getSerializedSvg() {
+    return getSerializedSvg(this)
+  }
+
+  getCommands() {
+    return this.recordedCommands
+  }
+
+  // setters (no getters working)
+  set strokeStyle(style: string) {
+    if (style !== this.currentStrokeStyle) {
+      this.pushSetterCall('strokeStyle', style)
+      this.currentStrokeStyle = style
+    }
+  }
+
+  set fillStyle(style: string) {
+    if (style !== this.currentFillStyle) {
+      this.pushSetterCall('fillStyle', style)
+      this.currentFillStyle = style
+    }
+  }
+
+  set font(style: string) {
+    this.currentFont = style
+    this.pushSetterCall('font', style)
+  }
+
+  // methods
+  arc(...args: RealP<'arc'>): RealRet<'arc'> {
+    this.pushMethodCall('arc', args)
+  }
+
+  arcTo(...args: RealP<'arcTo'>): RealRet<'arcTo'> {
+    this.pushMethodCall('arcTo', args)
+  }
+
+  beginPath(...args: RealP<'beginPath'>): RealRet<'beginPath'> {
+    this.pushMethodCall('beginPath', args)
+  }
+
+  clearRect(...args: RealP<'clearRect'>): RealRet<'clearRect'> {
+    this.pushMethodCall('clearRect', args)
+  }
+
+  closePath(...args: RealP<'closePath'>): RealRet<'closePath'> {
+    this.pushMethodCall('closePath', args)
+  }
+
+  ellipse(...args: RealP<'ellipse'>): RealRet<'ellipse'> {
+    this.pushMethodCall('ellipse', args)
+  }
+
+  fill(...args: RealP<'fill'>): RealRet<'fill'> {
+    this.pushMethodCall('fill', args)
+  }
+
+  fillRect(...args: RealP<'fillRect'>): RealRet<'fillRect'> {
+    const [x, y, w, h] = args
+    if (x > this.width || x + w < 0) {
+      return
+    }
+    const nx = Math.max(x, 0)
+    const nw = nx + w > this.width ? this.width - nx : w
```

Comment:
i think the nw here is not up to date with the current main branch, the current main does not have a ternary

```
    if (x > this.width || x + w < 0) {
      return
    }

    const nx = Math.max(x, 0)
    const nw = w - (nx - x)
```

Reply:
went ahead and pushed to this branch the fix, snapshot should be fixed